### ISSUE TITLE
version 3.15.12 not support mariadb returning (mariadb version above 10.5.0 )

### ISSUE THREAD

**Issue by JavanWang** [jOOQ/jOOQ#15027]
### Expected behavior

I am used to using jooq version 3.12. Recently, because of the demand for json functions, I upgraded the version to 3.15. But after the upgrade, I found that the **returning** function can no longer be used.

```
Exception                
org.jooq.exception.DataAccessException: SQL [insert into mydb.sample ( country_code,  type) values (?, ?) returning keyid]; Statement.executeQuery() cannot issue statements that do not produce result sets.
at org.jooq_3.15.12.MARIADB.debug(Unknown Source) ~[?:?]
at org.jooq.impl.Tools.translate(Tools.java:2997) ~[jooq-3.15.12.jar:?]
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:639) ~[jooq-3.15.12.jar:?]
```

The version of mariadb I am using is 10.11.2, https://mariadb.com/kb/en/insertreturning/ As you can see in the link, mariadb already supports return in 10.5, I can copy and paste the sql command generated by jooq to mariadb-client well executed
I looked back and compared the version 3.12. Jooq executes returning in segments, insert into --> select. 3.15 directly uses the function of returning, but it is not clear why the execution fails?

### Steps to reproduce the problem

this sql statement run well on mariadb
```
select version();
create table ocpidb.address
(id int auto_increment primary key,street varchar(10));
insert into address(street) values ('aaa') returning id;
```

### jOOQ Version

3.15.12

### Database product and version

10.11.2

### Java Version

11.0.18+10-post

### OS Version

Ubuntu-0ubuntu120.04.1

### JDBC driver name and version (include name if unofficial driver)

mysql.jdbc.version:8.0.30

**Comment by lukaseder** [comment]
Thanks a lot for your report. I take that you're using the mysql-connector JDBC driver?

> mysql.jdbc.version:8.0.30

We don't integration test that driver for the MariaDB database. Can you please use the `org.mariadb.jdbc:mariadb-java-client` driver instead?
https://central.sonatype.com/artifact/org.mariadb.jdbc/mariadb-java-client/3.1.4

**Comment by JavanWang** [comment]
@lukaseder Thank you very much for your quick reply, but the solution seem to be ineffective
I change dependency to mariadb-java-client, and jooq configuration jdbc modify to org.mariadb.jdbc.Driver, but the problem still exists.
I tried to find a workaround to avoid this problem
org.jooq.Configuration conf = new DefaultConfiguration().set(SQLDialect.**MARIADB**)
change to 
org.jooq.Configuration conf = new DefaultConfiguration().set(SQLDialect.**MYSQL**)
everything is ok

**Comment by lukaseder** [comment]
Yeah, `MYSQL` doesn't support the `RETURNING` clause, so changing dialects "fixes" that, but will open up tons of other problems.

How can I reproduce this? Our integration tests obviously don't show such a big problem. Can you provide an MCVE (template here: https://github.com/jOOQ/jOOQ-mcve). Or at least the complete stack trace?

**Comment by JavanWang** [comment]
@lukaseder  Sorry for bothering you with this issue. 

I have figured out what the problem was. I changed the Maven POM file jOOQ-jdbc from _com.mysql.cj.jdbc.Driver_ to _org.mariadb.jdbc.Driver_, but I didn't notice that jOOQ depends on HikariDataSource settings. Therefore, the HikariConfig was still set to jdbc:mysql://... 

Once I changed the HikariDataSource config to jdbc:mariadb://..., everything ran well. Thank you very much.
